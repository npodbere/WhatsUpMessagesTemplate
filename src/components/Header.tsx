import { ReactComponent as Close } from "../img/Close.svg";
import { ReactComponent as FlashLight } from "../img/FlashLight.svg";

interface IHeader {
  title: string;
}

function Header({ title }: IHeader) {
  return (
    <div className="header-container">
      <div className="title-small">{title}</div>
      <div className="header-right-container">
        <div className="header-tips-container">
          <FlashLight />
          <div className="button-blue-small">Tips</div>
        </div>
        <div style={{ cursor: "pointer" }}>
          <Close />
        </div>
      </div>
    </div>
  );
}

export default Header;
