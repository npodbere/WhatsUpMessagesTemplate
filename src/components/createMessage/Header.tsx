import Tips from "../common/Tips";

function Header() {
  return (
    <>
      <select className="create-message-header-select">
        <option value="someOption">Image</option>
      </select>
      <p style={{ whiteSpace: "nowrap" }}>
        Image size recommendation: 800 x 418 pixel.
      </p>
      <button
        className="create-message-header-upload-btn button-blue-large"
        type="button"
      >
        Upload Image
      </button>
      <Tips
        title="Image header tips"
        text="Images can enrich the message experience and help maintain engagement. Use eye-catching images that summarize the message (eg discounts, gifts etc.)"
      />
    </>
  );
}

export default Header;
