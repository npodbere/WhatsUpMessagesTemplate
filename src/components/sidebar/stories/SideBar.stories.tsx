import type { Meta, ReactRenderer, StoryObj } from "@storybook/react";
import { BrowserRouter } from "react-router-dom";
import type { PartialStoryFn } from "@storybook/types";

import SideBar from "../SideBar";

const meta = {
  component: SideBar,
  decorators: [
    (Story: PartialStoryFn<ReactRenderer>): JSX.Element => (
      <BrowserRouter>
        <div style={{ display: "flex", height: "100vh" }}>
          <Story />
        </div>
      </BrowserRouter>
    ),
  ],
  parameters: {
    layout: "fullscreen",
  },
  title: "sidebar/SideBar",
} satisfies Meta<typeof SideBar>;

export default meta;
type Story = StoryObj<typeof meta>;

export const initialRendering: Story = {};
