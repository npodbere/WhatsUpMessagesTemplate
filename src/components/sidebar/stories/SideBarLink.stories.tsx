import type { Meta, ReactRenderer, StoryObj } from "@storybook/react";
import { BrowserRouter } from "react-router-dom";
import type { PartialStoryFn } from "@storybook/types";

import { ReactComponent as Dashboard } from "../../../img/dashboard.svg";
import SideBarLink from "../SideBarLink";

const meta = {
  component: SideBarLink,
  decorators: [
    (Story: PartialStoryFn<ReactRenderer>): JSX.Element => (
      <BrowserRouter>
        <div style={{ backgroundColor: "#000" }}>
          <Story />
        </div>
      </BrowserRouter>
    ),
  ],
  parameters: {
    layout: "fullscreen",
  },
  title: "sidebar/SideBarLink",
} satisfies Meta<typeof SideBarLink>;

export default meta;
type Story = StoryObj<typeof meta>;

export const initialRendering: Story = {
  args: {
    Icon: Dashboard,
    destinationPath: "dashboard",
  },
};
