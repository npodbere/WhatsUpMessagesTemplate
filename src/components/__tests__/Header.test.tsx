import { create } from "react-test-renderer";
import { composeStories } from "@storybook/react";

import * as HeaderStories from "../stories/Header.stories";

const { InitialRendering } = composeStories(HeaderStories);

test("initial rendering of the header", async () => {
  const tree = create(<InitialRendering />);

  expect(tree).toMatchSnapshot();
});
