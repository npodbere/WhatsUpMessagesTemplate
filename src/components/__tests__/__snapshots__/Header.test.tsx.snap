// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`initial rendering of the header 1`] = `
<div
  className="header-container"
>
  <div
    className="title-small"
  >
    Header Title
  </div>
  <div
    className="header-right-container"
  >
    <div
      className="header-tips-container"
    >
      <svg>
        FlashLight.svg
      </svg>
      <div
        className="button-blue-small"
      >
        Tips
      </div>
    </div>
    <div
      style={
        Object {
          "cursor": "pointer",
        }
      }
    >
      <svg>
        Close.svg
      </svg>
    </div>
  </div>
</div>
`;
